File: library/src/main/java/com/pengrad/telegrambot/model/GameHighScore.java
package com.pengrad.telegrambot.model;

import java.io.Serializable;
import java.util.Objects;

/**
 * Stas Parshin
 * 04 October 2016
 */
public class GameHighScore implements Serializable {
    private final static long serialVersionUID = 0L;

    private Integer position;
    private User user;
    private Integer score;

    public Integer position() {
        return position;
    }

    public User user() {
        return user;
    }

    public Integer score() {
        return score;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        GameHighScore that = (GameHighScore) o;

        if (!Objects.equals(position, that.position)) return false;
        if (!Objects.equals(user, that.user)) return false;
        return Objects.equals(score, that.score);
    }

    @Override
    public int hashCode() {
        int result = position != null ? position.hashCode() : 0;
        result = 31 * result + (user != null ? user.hashCode() : 0);
        result = 31 * result + (score != null ? score.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "GameHighScore{" +
                "position=" + position +
                ", user=" + user +
                ", score=" + score +
                '}';
    }
}

File: library/src/main/java/com/pengrad/telegrambot/model/GameHighScore.kt
package com.pengrad.telegrambot.model

import java.io.Serializable

/**
 * Stas Parshin
 * 04 October 2016
 */
data class GameHighScore(
        @get:JvmName("position") val position: Int? = null,
        @get:JvmName("user") val user: User? = null,
        @get:JvmName("score") val score: Int? = null
) : Serializable {
    companion object {
        private const val serialVersionUID = 0L
    }
}